app/jobs/test_track/assignment_event_job.rb

```diff
@@ -0,0 +1,71 @@
+module TestTrack
+  class AssignmentEventJob < TestTrack.job_base_class_name.constantize
+    attr_reader :visitor_id, :assignment
+
+    def perform(visitor_id:, assignment:)
+      raise "visitor_id must be present" if visitor_id.blank?
+      raise "assignment must be present" if assignment.blank?
+
+      @visitor_id = visitor_id
+      @assignment = build_assignment(visitor_id, assignment)
```

Comment:
This `build_assignment` step is the one new thing about this class. The `TestTrack::Assignment` model is not serializable by ActiveJob, and rather than writing a custom serializer, I pass its fields through as a Hash and then reconstruct it inside of the job.